frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,10 +1,75 @@
-type Properties = {
-    label: string;
-    type?: 'button' | 'submit';
+import clsx from 'clsx';
+
+import {
+    type ButtonTheme,
+    ButtonSize,
+    IconName,
+    IconSize,
+} from '~/bundles/common/enums/enums';
+import { type ButtonType, type ValueOf } from '~/bundles/common/types/types';
+
+import { Icon } from '../icon/icon';
+import styles from './styles.module.scss';
+
+type ButtonProperties = {
+    label?: string;
+    type?: ButtonType;
+    theme?: ValueOf<typeof ButtonTheme>;
```

Comment:
Consider renaming this prop to `variant`, because usually theme is about interface colors, (e.g. `light`, `dark`)